### PR TITLE
fix/test(CL Fees): swap out given in fee growth tests; clean up swap tests

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

Due to a merge conflict between https://github.com/osmosis-labs/osmosis/pull/4021 and https://github.com/osmosis-labs/osmosis/pull/4097, the fee growth validations were not added.

This PR adds the validations for the fee growth accumulator.

The changes from  https://github.com/osmosis-labs/osmosis/pull/4021 interfere with the assertions since they also relate to updating the fee growth accumulator. As a result, they were moved into a separate test fixture called `TestSwapOutAmtGivenIn_TickUpdates` to enable the expected fee growth validations in the original test.

Lastly, it cleans up the swap test suite. See changelog for details. This clean up is necessary for https://github.com/osmosis-labs/osmosis/pull/4131 and is separated here for ease of review.

## Brief Changelog

- move tick update tests into a separate fixture `TestSwapOutAmtGivenIn_TickUpdates`
- enable validation of the fee growth accumulator in `TestCalcAndSwapOutAmtGivenIn`
- do the first two steps for swap in given out in preparation for https://github.com/osmosis-labs/osmosis/pull/4131
- group fee-specific tests in `swapOutGivenInFeeCases` global variable
  * this separation stems from the need to avoid running fee tests in tick updates or inverse tests
- group all test case fields in struct `SwapTest` and reuse it where applicable

## Testing and Verifying

This change is already covered by existing tests

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not applicable

### CODE REVIEW

File: x/concentrated-liquidity/swaps_test.go [hunk 1]
```diff
@@ -581,232 +560,28 @@ var (
 			expectErr:     true,
 		},
 	}
-)
-
-func (s *KeeperTestSuite) TestCalcAndSwapOutAmtGivenIn() {
```
Comment (p0mvn) [inline x/concentrated-liquidity/swaps_test.go:586]: Note: this is a problem with diff, the test is present down below

File: x/concentrated-liquidity/swaps_test.go [hunk 2]
```diff
@@ -1117,69 +791,311 @@ func (s *KeeperTestSuite) TestCalcAndSwapInAmtGivenOut() {
...
 	for name, test := range tests {
+		test := test
 		s.Run(name, func() {
 			s.Setup()
 			s.FundAcc(s.TestAccs[0], sdk.NewCoins(sdk.NewCoin("eth", sdk.NewInt(10000000000000)), sdk.NewCoin("usdc", sdk.NewInt(1000000000000))))
 			s.FundAcc(s.TestAccs[1], sdk.NewCoins(sdk.NewCoin("eth", sdk.NewInt(10000000000000)), sdk.NewCoin("usdc", sdk.NewInt(1000000000000))))
 
 			// Create default CL pool
-			pool := s.PrepareConcentratedPool()
+			pool := s.PrepareCustomConcentratedPool(s.TestAccs[0], ETH, USDC, DefaultTickSpacing, DefaultExponentAtPriceOne, sdk.ZeroDec())
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/swaps_test.go:853]: Was the custom CL pool func needed here to specify the testAcc?

**Review by czarcas7ic — APPROVED** [review]
Approved with the understanding that we are adding swapInGivenOutFeeCases post fee logic PR. Nicely done, very clean (the diff was super odd but reviewing the final file instead was much easier to grok)

File: x/concentrated-liquidity/swaps_test.go (hunk 2 above)
Comment (p0mvn) [reply x/concentrated-liquidity/swaps_test.go:853]: No, will change back in the next PR

Comment (p0mvn) [reply x/concentrated-liquidity/swaps_test.go]: fixed now

**Comment by p0mvn** [comment]
merging since this is a simple test refactor

**Pull request merged**